### PR TITLE
Fixed issue #19094: Input Validation Vulnerability Leading to 500 on responses page

### PR DESCRIPTION
**Pull request by Shnoulle**
Dev: check size validity before use
Dev: filter when save

Fixed issue #<Mantis issue number>:
New feature #<Mantis issue number>:
Dev:

### CODE REVIEW

File: application/models/SurveyDynamic.php [hunk 1]
```diff
@@ -404,7 +404,7 @@ public function getExtendedData($colName, $sLanguage, $base64jsonFieldMap)
                 $sSurveyEntry .= '<tr>';
                 if (isset($aFilesInfo[$iFileIndex])) {
                     $sSurveyEntry .= '<td>'.CHtml::link(CHtml::encode(rawurldecode($aFilesInfo[$iFileIndex]['name'])), App()->createUrl("/admin/responses", array("sa"=>"actionDownloadfile", "surveyid"=>self::$sid, "iResponseId"=>$this->id, "iQID"=>$oFieldMap->qid, "iIndex"=>$iFileIndex))).'</td>';
-                    $sSurveyEntry .= '<td>'.sprintf('%s Mb', round($aFilesInfo[$iFileIndex]['size'] / 1000, 2)).'</td>';
+                    $sSurveyEntry .= '<td>' . (($aFilesInfo[$iFileIndex]['size'] && is_float($aFilesInfo[$iFileIndex]['size'])) ? sprintf('%s Mb', round($aFilesInfo[$iFileIndex]['size'] / 1000, 2)) : "") . '</td>';
```
Comment (gabrieljenik) [inline application/models/SurveyDynamic.php:407]: Not too much inlove with inline-code, but that's a personal taste I guess, right?

**Review by gabrieljenik — APPROVED** [review]
Code looks OK

File: application/models/SurveyDynamic.php (hunk 1 above)
Comment (Shnoulle) [reply application/models/SurveyDynamic.php:407]: I do the quickest way here (3X). Maybe some uodate for 5 or 6 then ?

Comment (Shnoulle) [reply application/models/SurveyDynamic.php:407]: See the previous celle (whu it's a cell ????)

`'<td>'.CHtml::link(CHtml::encode(rawurldecode($aFilesInfo[$iFileIndex]['name'])), App()->createUrl("/admin/responses", array("sa"=>"actionDownloadfile", "surveyid"=>self::$sid, "iResponseId"=>$this->id, "iQID"=>$oFieldMap->qid, "iIndex"=>$iFileIndex))).'</td>';`

**Pull request merged**